components/application_manager/hbbtv_app.h

```diff
@@ -43,7 +42,11 @@ class App
     typedef enum
     {
         FOREGROUND_STATE = 1,
-        BACKGROUND_STATE = 1 << 1
+        BACKGROUND_STATE = 1 << 1,
+        TRANSIENT_STATE = 1 << 2,
+        OVERLAID = 1 << 3, // only for use with other states as below
+        OVERLAID_FOREGROUND_STATE = OVERLAID | FOREGROUND_STATE,
+        OVERLAID_TRANSIENT_STATE = OVERLAID | TRANSIENT_STATE
```

Comment:
As the code is, there are no simultaneous states at any given point. The bitmask was introduced to avoid multiple ORs and ANDs inside the canTransitionToState() method.
With regards to the OVERLAID state, it is only present in the enum and not an actual state of an OpApp, and is meant to be piped with either FORGROUND or TRANSIENT state.

Reply:
OK, so if you can just go with removing OVERLAID state and use a straightforward, discrete state model that will things easier.

{ 
  FOREGROUND _STATE=1, 
  BACKGROUND_STATE, 
  TRANSIENT_STATE, 
  OVERLAID_FOREGROUND_STATE, 
  OVERLAID_TRANSIENT_STATE,
  ILLEGAL_STATE
 };